hnn_core/viz.py

```diff
@@ -242,29 +244,45 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    from .dipole import Dipole
+    from .dipole import Dipole, average_dipoles
 
     if ax is None:
         _, ax = plt.subplots(1, 1, constrained_layout=True)
 
     if isinstance(dpl, Dipole):
         dpl = [dpl]
+        average = False
+    else:
+        # add average plot when we have dpl>1 and `average` is specified.
+        if average:
+            dpl = dpl + [average_dipoles(dpl)]
 
+    linewidth = 1.5
     scale_applied = dpl[0].scale_applied
-    for dpl_trial in dpl:
+    for idx, dpl_trial in enumerate(dpl):
         if dpl_trial.scale_applied != scale_applied:
             raise RuntimeError('All dipoles must be scaled equally!')
 
         if layer in dpl_trial.data.keys():
 
             # extract scaled data and times
             data, times = _get_plot_data_trange(dpl_trial.times,
-                                                dpl_trial.data[layer],
-                                                tmin, tmax)
+                                                dpl_trial.data[layer], tmin,
+                                                tmax)
             if decim is not None:
                 data, times = _decimate_plot_data(decim, data, times)
 
-            ax.plot(times, data, color=color)
+            if idx == len(dpl) - 1 and average:
+                # the last one (average dpl) or the single dpl
```

Comment:
```suggestion
                # the last one (average dpl)
```

Reply:
mixing the two will cause problems. Hence, see what I [suggested above](https://github.com/jonescompneurolab/hnn-core/pull/475/files#r835728105)